networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/Eth2PeerManager.java

```diff
@@ -105,7 +111,49 @@ public static Eth2PeerManager create(
         peerValidatorFactory,
         attestationSubnetService,
         rpcEncoding,
-        eth2RpcPingInterval);
+        eth2RpcPingInterval,
+        eth2StatusUpdateInterval);
+  }
+
+  private void setUpPeriodicTasksForPeer(Eth2Peer peer) {
+    SafeFuture<Void> periodicStatusUpdateTask = periodicallyUpdatePeerStatus(peer);
+    SafeFuture<Void> periodicPingTask = periodicallyPingPeer(peer);
+    peer.subscribeDisconnect(() -> periodicStatusUpdateTask.cancel(false));
+    peer.subscribeDisconnect(() -> periodicPingTask.cancel(false));
```

Comment:
Suggest combining these into one call to `subscribeDisconnect`:
```suggestion
    peer.subscribeDisconnect(() -> {
        periodicStatusUpdateTask.cancel(false);
        periodicPingTask.cancel(false);
     });
```

Reply:
Done.